Project 4/README.md

```diff
@@ -45,3 +45,133 @@ You're probably thinking, "why would I submit a level 8 kata if they're not wort
 It is your responsibility and the responsibility of your peers reviewing your submission in PR to determine whether your submission is ranked appropriately. In the event that consensus is reached that your kata is ranked inappropriately, you must work with your peers to revise the submission so that it is either more or less challenging, accordingly. You are not permitted to submit new problems with different strengths after PRs are open, but must instead revise your PRs. So, think hard about how challenging your submission is. 
 
 There is one other option for those desiring a different sort of challenge. If you provide alongside your SPARQL submission a translation of the same problem into SQL, complete with documentations, solution, etc. then you may receive half points extra at that kata level (rounded up). For example, if you submit a SPARQL problem that is kata rank 1 and also submit a SQL version of that same problem, you  will receive 35+18=53 points. 
+
+My Best Attempts.....
+Kata Level 6 – 3 point
+All actors who have played the character of Batmen.
+PREFIX dbo: http://dbpedie.org/ontology/
+PREFIX dbr: http://dbpedia.org/research/
+Select Distinct ?actor WHERE {
+?film dbo:starring ?act .
+?film dbo:wikiPageRedircts dbr:Batman_(film_series) . 
+?actor dbo:wikiPageID ?id . 
+FILTER (REGEX(STR(?id), “Batman_films_casts”, “I”))
+} 
+
+Kata Level 6 – 6 points
+Who invented the telephone?
+PREFIX dbo: http://dbpedia.org/ontology/
+PREFIX foaf: http://xmlns.com/foaf/0.1/
+Select ?name
+?telephone a dbo:Device ; 
+	Dbo:inventor ?inventor .
+?inventor foaf:name ?name . 
+Filter (regex(?name, “Bell”, “I”))
+}
+
+Kata level 8 – 0 points
+Return all classes  -- (A is shorthand for rdfs; type)
+SELECT DISTINCT ?class
+WHERE {
+	?s a ?class .
+FILTER (strstarts(str(?class), http://dbpedia.org/ontology/))
+}
+Kata level 5 – 11 points
+Write a SPARQL query that CONSTRUCTS a new graph of all books that have been written by authors who have also written books in the “Science Fiction” genre. The new graph should include the title, author, and genre of each book, as well as the name and birth year if each author.
+PREFIX rdf: http://www.w3.org/1999/02/22-rdf-syntax-ns#
+PREFIX dc: http://purl.org/dc/terms/
+PREFIX foaf: http://xmlns.com/foaf/0.1/
+PREFIX genre http://example.com/genre#
+
+CONSTRUCT {
+ ?book rdf:type dc:Book ;
+dc:title >?title ;
+dc:creator ?author ;
+genre:ScienceFiction ?genre . 
+?author rdf: type foaf:Person ;
+	foaf:name ?name ;
+	foaf:birthyear ?birthYear . 
+}
+WHERE{
+?book rdf:type dc:Book ;
+	dc:title ?title
+	dc:creator ?author ;
+	genre:Sciencefiction ?genre . 
+?author rdf:type foaf:Person ;
+	foaf:name ?name ; 
+	foaf:birthyear ?birthYear . 
+FILTER EXISTS {
+?otherBook rdf:type dc:Book ;
+	dc:creator ?author ; 
+genre:ScienceFicton ?genre . 
+Filter (?otherBook !=?book)
```

Comment:
Let's say that an author has written only one book, of genre science fiction. I wonder whether it would fall outside of the scope of this query, because it doesn't have an ?otherbook. 
It is probably easier to rewrite the question than to rewrite the query. As I see it now, it seems to ask for all the authors that have at least two books, at least one of which is a sci fi book.

Reply:
I had not considered this. Thank you for pointing this out I will rewrite the question